drivers/sensors/sensor.c

```diff
@@ -801,9 +654,9 @@ int sensor_register(FAR struct sensor_lowerhalf_s *lower, int devno)
 
   /* Initialize sensor buffer */
 
-  ret = sensor_buffer_create(&upper->buffer,
-                             lower->type, lower->buffer_size);
-  if (ret)
+  bytes = ROUNDUP(lower->buffer_size, g_sensor_info[lower->type].esize);
```

Comment:
Same concern as in previous PR: when using only fetch, won't this allocate unneeded memory? Why is it needed if fetch will write directly to user buffer?

Reply:
When fetch is NULL, the lower->buffer_size is zero, and ROUNDUP(lower->buffer_size,...) also is zero, so circ_buffer_init will not alloc buffer space for sensor using fetch api. This ensures code consistency between not fetch and use fetch.

Reply:
Ok, I was confused with the purpose of ROUNDUP. I assumed it would return `g_sensor_info[lower->type].esize` in that case. If it returns 0 it is OK.